internal/ingress/annotations/authtls/main.go

```diff
@@ -130,9 +130,14 @@ func (a authTLS) Parse(ing *networking.Ingress) (interface{}, error) {
 		config.PassCertToUpstream = false
 	}
 
-	config.MatchCN, err = parser.GetStringAnnotation("auth-tls-match-cn", ing)
-	if err != nil || !commonNameRegex.MatchString(config.MatchCN) {
-		config.MatchCN = ""
+	config.ExactMatchCN, err = parser.GetStringAnnotation("auth-tls-exact-match-cn", ing)
```

Comment:
There should be no priority, it is up to the user which annotation they want to use!

Reply:
I see what you mean, I think because they are 2 separate annotations then it's hard to prioritize one over the other.
because the original is called auth-tls-match-cn, maybe we can check that one first and if it's not empty then we can exit early. If it is empty we can then check the auth-tls-exact-match-cn

Reply:
since they are essentially 2 different annotations there isn't necessarily a priority.